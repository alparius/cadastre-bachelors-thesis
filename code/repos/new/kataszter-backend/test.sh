#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
