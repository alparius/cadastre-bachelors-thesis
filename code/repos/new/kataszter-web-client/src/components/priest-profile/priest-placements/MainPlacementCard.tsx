import React from "react";
import { NavLink } from "react-router-dom";
import { Button, Card, Header, List } from "semantic-ui-react";

import IParishMin from "../../../data/parish/ParishMin";
import IMainPlacement from "../../../data/priest/MainPlacement";
import "../../../static/priestProfile.css";
import CardRow from "../CardRow";
import CardRowForDate from "../CardRowForDate";
import CardRowPlacementDropDown from "./CardRowPlacementDropDown";

interface IMainPlacementCardProps {
    placement: IMainPlacement;
    index: number;
    editMode: boolean;
    handleDeletePlacement: (index: number) => void;
    handleChangeOnPlacementField: (labelName: string, value: any, index: number) => void;
    handlePlacementChange: (placeID: number, place: string, index: number) => void;
    parishResponse: IParishMin[] | undefined;
    parishLoading: boolean;
    parishShowError: boolean;
}

const MainPlacementCard: React.FC<IMainPlacementCardProps> = (props: IMainPlacementCardProps) => {
    const {
        placement,
        index,
        editMode,
        handleDeletePlacement,
        handleChangeOnPlacementField,
        handlePlacementChange,
        parishResponse,
        parishLoading,
        parishShowError
    } = props;

    const editHandlers = {
        index,
        editMode,
        setValue: handleChangeOnPlacementField
    };

    const handleDeletePressed = () => {
        handleDeletePlacement(index);
    };

    const getCardContent = () => {
        return (
            <Card.Content>
                <Card.Content>
                    {editMode ? (
                        <Button id="deleteButton" floated="right" icon="trash" color="red" inverted onClick={handleDeletePressed} />
                    ) : (
                        <Card.Header id="header">
                            <Header as="h2">{placement.place}</Header>
                        </Card.Header>
                    )}
                </Card.Content>

                <Card.Description>
                    <List>
                        {editMode && (
                            <List.Item>
                                <CardRowPlacementDropDown
                                    placeID={placement.placeID}
                                    placeName={placement.place}
                                    index={index}
                                    editMode={editMode}
                                    handlePlacementChange={handlePlacementChange}
                                    parishResponse={parishResponse}
                                    parishLoading={parishLoading}
                                    parishShowError={parishShowError}
                                />
                            </List.Item>
                        )}
                        <List.Item>
                            <CardRowForDate labelName="startDate" data={placement.startDate} {...editHandlers} />
                        </List.Item>
                        <List.Item>
                            <CardRowForDate labelName="endDate" data={placement.endDate} {...editHandlers} />
                        </List.Item>
                        <List.Item>
                            <CardRow labelName="numberOfPeople" data={placement.numberOfPeople} {...editHandlers} />
                        </List.Item>
                        <List.Item>
                            <CardRow labelName="munimentIssuingAuthority" data={placement.munimentIssuingAuthority} {...editHandlers} />
                        </List.Item>
                        <List.Item>
                            <CardRowForDate labelName="munimentGenesisDate" data={placement.munimentGenesisDate} {...editHandlers} />
                        </List.Item>
                        <List.Item>
                            <CardRow labelName="munimentNumber" data={placement.munimentNumber} {...editHandlers} />
                        </List.Item>
                        <List.Item>
                            <CardRow labelName="income" data={placement.income} {...editHandlers} />
                        </List.Item>
                    </List>
                </Card.Description>
            </Card.Content>
        );
    };

    if (editMode) {
        return <Card fluid>{getCardContent()}</Card>;
    } else {
        return (
            <Card fluid as={NavLink} to={`/parishes/profile/${placement.placeID}`}>
                {getCardContent()}
            </Card>
        );
    }
};

export default MainPlacementCard;
