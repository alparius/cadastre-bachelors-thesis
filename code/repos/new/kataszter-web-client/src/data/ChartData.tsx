interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
