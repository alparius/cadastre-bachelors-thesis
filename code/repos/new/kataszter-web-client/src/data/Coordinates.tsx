interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
