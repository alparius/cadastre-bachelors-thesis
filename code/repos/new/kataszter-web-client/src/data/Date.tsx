interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
