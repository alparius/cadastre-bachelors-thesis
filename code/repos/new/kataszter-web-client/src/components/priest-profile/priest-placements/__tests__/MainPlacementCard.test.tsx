/*===================
  MainPlacementCard.unit.test.tsx
 ===================*/
import React from "react";

import { NewTestMainPlacement } from "../../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter, mountWithIntlAndRouter } from "../../../../util/test/test-wrapper";
import MainPlacementCard from "../MainPlacementCard";

const handlers = {
    handleDeletePlacement: jest.fn(),
    handleChangeOnPlacementField: jest.fn(),
    handlePlacementChange: jest.fn(),
    parishResponse: undefined,
    parishLoading: false,
    parishShowError: false
};

describe("MainPlacementCard element", () => {
    afterEach(() => {
        handlers.handleDeletePlacement.mockClear();
        handlers.handleChangeOnPlacementField.mockClear();
        handlers.handlePlacementChange.mockClear();
    });

    it("should render a card with the placement data", () => {
        const container = createComponentWithIntlAndRouter(
            <MainPlacementCard placement={NewTestMainPlacement} index={0} editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call handleDeletePlacement if deleteButton pressed", () => {
        const container = mountWithIntlAndRouter(<MainPlacementCard placement={NewTestMainPlacement} index={0} editMode {...handlers} />);
        container
            .find("#deleteButton")
            .at(0)
            .simulate("click");

        expect(handlers.handleDeletePlacement).toHaveBeenCalledTimes(1);
    });

    it.each([
        [1, "1", "2"],
        [2, "1", "2"],
        [3, "1", "2"],
        [4, "1", "2"],
        [5, "1", "2"],
        [6, "1", "2"],
        [7, "1", "2"]
    ])(
        "should  call handleChangeOnPlacementField on %d. input. After first change: %s, After second change: %s",
        (inputNr, firstChange, secondChange) => {
            const container = mountWithIntlAndRouter(<MainPlacementCard placement={NewTestMainPlacement} index={0} editMode {...handlers} />);
            const input = container.find("input").at(Number(inputNr));
            input.simulate("change", { target: { value: firstChange } });
            input.simulate("change", { target: { value: secondChange } });
            expect(handlers.handleChangeOnPlacementField).toHaveBeenCalledTimes(2);
        }
    );
});
