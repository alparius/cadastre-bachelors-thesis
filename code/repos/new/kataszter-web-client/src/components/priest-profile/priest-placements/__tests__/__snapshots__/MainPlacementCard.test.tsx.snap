// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPlacementCard element should render a card with the placement data 1`] = `
<a
  aria-current={null}
  className="ui fluid card"
  href="/parishes/profile/0"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="content"
    >
      <div
        className="header"
        id="header"
      >
        <h2
          className="ui header"
        >
          place
        </h2>
      </div>
    </div>
    <div
      className="description"
    >
      <div
        className="ui list"
        role="list"
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Start date
                :
              </b>
               
              <span
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                2000.06.06
              </span>
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                End date
                :
              </b>
               
              <span
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                2000.06.06
              </span>
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Number of people
                :
              </b>
               
              numberOfPeople
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Muniment issuing authority
                :
              </b>
               
              munimentIssuingAuthority
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Muniment genesis date
                :
              </b>
               
              <span
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                2000.06.06
              </span>
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Muniment number
                :
              </b>
               
              munimentNumber
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Income
                :
              </b>
               
              income
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</a>
`;
